composer/trainer/devices/device_tpu.py

```diff
@@ -0,0 +1,50 @@
+# Copyright 2022 MosaicML Composer authors
+# SPDX-License-Identifier: Apache-2.0
+
+"""The TPU device used for training."""
+
+from __future__ import annotations
+
+import logging
+from typing import Any, Dict, TypeVar
+
+import torch
+
+from composer.trainer.devices.device import Device
+from composer.utils import MissingConditionalImportError
+
+logger = logging.getLogger(__name__)
+
+__all__ = ['DeviceTPU']
+
+T_nnModule = TypeVar('T_nnModule', bound=torch.nn.Module)
+
+
+class DeviceTPU(Device):
+    """An extension of :class:`~composer.trainer.devices.device.Device` for TPUs.
+
+    When running on TPUVMs, you need to `export PJRT_DEVICE=TPU`.
+    More details.
+    """
+
+    def __init__(self):
+        try:
+            import torch_xla.core.xla_model as xm
+        except ImportError as e:
+            raise MissingConditionalImportError(extra_deps_group='tpu', conda_package='torch_xla[tpuvm]') from e
```

Comment:
I don't see a `tpu` group in `setup.py`, and does `conda install torch_xla[tpuvm]` actually work?